compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BReadonlyType.java

```diff
@@ -44,14 +45,14 @@ private BReadonlyType(BTypeSymbol tsymbol, SemType semType) {
     }
 
     public BReadonlyType(BTypeSymbol tsymbol, Name name, long flag) {
-        super(TypeTags.READONLY, tsymbol, SemTypeHelper.READONLY_SEMTYPE);
+        super(TypeTags.READONLY, tsymbol, VAL_READONLY);
         this.name = name;
         this.flags = flag;
         this.flags |= Flags.READONLY;
     }
 
     public static BReadonlyType newNilLiftedBReadonlyType(BTypeSymbol tsymbol) {
-        return new BReadonlyType(tsymbol, Core.diff(SemTypeHelper.READONLY_SEMTYPE, PredefinedType.NIL));
+        return new BReadonlyType(tsymbol, Core.diff(VAL_READONLY, PredefinedType.NIL));
```

Comment:
can override `semtype()` to return `VAL_READONLY` for this class?

Reply:
Not entirely sure that is correct due to this `newNilLiftedBReadonlyType` method.  My understanding is that this method need to give a `ReadonlyType` which don't include `nil` and if we override the `semtype()` method we'll return a semtype that include `nil` even for these "nil lifted readonly" values. Also remove the `semtype()` override from `BAnyType` as well for the same reason.